diesel_cli/src/cli.rs

```diff
@@ -18,11 +18,41 @@ pub fn build_cli() -> App<'static, 'static> {
         .setting(AppSettings::VersionlessSubcommands)
         .arg(migration_dir_arg())
         .subcommand(SubCommand::with_name("run").about("Runs all pending migrations"))
-        .subcommand(SubCommand::with_name("revert").about("Reverts the latest run migration"))
-        .subcommand(SubCommand::with_name("redo").about(
-            "Reverts and re-runs the latest migration. Useful \
-             for testing that a migration can in fact be reverted.",
-        ))
+        .subcommand(
+            SubCommand::with_name("revert")
+                .about("Reverts the latest run migration")
+                .arg(
+                    Arg::with_name("REVERT_NUMBER")
+                        .long("number")
+                        .short("n")
+                        .help("Reverts the last migration files")
+                        .long_help(
+                            "When this option is specified the last `n` migration files \
+                             will be reverted. By default revert the last one.",
+                        )
+                        .takes_value(true)
+                        .required(true)
+                        .default_value("1")
+                        .validator(migration_revert_number_validator),
+                ),
+        )
+        .subcommand(
+            SubCommand::with_name("redo")
```

Comment:
This subcommand should have a `number` argument which should default to `1` and should be exclusive with `all` arg.

Reply:
Hello @pksunkara 

I tried to use `conflicts_with` and a `default_value`, but it doesn't work since the default value will "activate" the `Arg`. 

```rust
.subcommand(
            SubCommand::with_name("revert")
                .about("Reverts the specified migrations")
                .arg(
                    Arg::with_name("REVERT_ALL")
                        .long("all")
                        .short("a")
                        .help("Reverts previously run migration files.")
                        .takes_value(false)
                        .conflicts_with("REVERT_NUMBER"),
                )
                .arg(
                    Arg::with_name("REVERT_NUMBER")
                        .long("number")
                        .short("n")
                        .help("Reverts the last `n` migration files")
                        .long_help(
                            "When this option is specified the last `n` migration files \
                             will be reverted. By default revert the last one.",
                        )
                        .takes_value(true)
                        .default_value("1")
                        .validator(is_positive_int)
                        .conflicts_with("REVERT_ALL"),
                ),
```

This will throw an error : 

```
diesel migration revert --all
stderr: error: The argument '--all' cannot be used with '--number <REVERT_NUMBER>'
```

If you need it, I can push my current code with the failing test.